pkg/sql/create_external_connection.go

```diff
@@ -112,21 +114,27 @@ func (p *planner) createExternalConnection(
 	ex.SetConnectionType(exConn.ConnectionType())
 	ex.SetOwner(p.User())
 
-	// Create the External Connection and persist it in the
-	// `system.external_connections` table.
-	if err := ex.Create(params.ctx, params.ExecCfg().InternalExecutor, params.p.User(), p.Txn()); err != nil {
-		return errors.Wrap(err, "failed to create external connection")
+	if err := p.WithInternalExecutor(params.ctx, func(ctx context.Context, txn *kv.Txn, ie sqlutil.InternalExecutor) error {
```

Comment:
nit: could just be `return p.WithInternalExecutor(...`

Reply:
Done.